assets/src/edit-story/components/canvas/utils/useSnapping.js

```diff
@@ -14,34 +14,100 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+
 /**
  * Internal dependencies
  */
 import { useGlobalIsKeyPressed } from '../../keyboard';
 import useCanvas from '../useCanvas';
 
 function useSnapping({ canSnap, otherNodes }) {
-  const { canvasWidth, canvasHeight } = useCanvas(
+  const {
+    canvasWidth,
+    canvasHeight,
+    pageContainer,
+    canvasContainer,
+    designSpaceGuideline,
+    showSafeZone,
+  } = useCanvas(
     ({
       state: {
         pageSize: { width: canvasWidth, height: canvasHeight },
+        pageContainer,
+        canvasContainer,
+        designSpaceGuideline,
+        showSafeZone,
       },
-    }) => ({ canvasWidth, canvasHeight })
+    }) => ({
+      canvasWidth,
+      canvasHeight,
+      pageContainer,
+      canvasContainer,
+      designSpaceGuideline,
+      showSafeZone,
+    })
   );
 
   // ⌘ key disables snapping
   const snapDisabled = useGlobalIsKeyPressed('meta');
   canSnap = canSnap && !snapDisabled;
+
+  const toggleDesignSpace = useCallback(
+    (visible) => {
+      designSpaceGuideline.style.visibility = visible ? 'visible' : 'hidden';
+    },
+    [designSpaceGuideline]
+  );
+  const handleSnap = useCallback(
+    ({ elements }) =>
+      // Show design space if we're snapping to any of its edges
+      toggleDesignSpace(
+        elements
+          .flat()
+          .some(
+            ({ center, element }) => element === designSpaceGuideline && !center
+          )
+      ),
+    [toggleDesignSpace, designSpaceGuideline]
+  );
+
+  if (!canvasContainer || !pageContainer) {
+    return {};
+  }
+
+  const canvasRect = canvasContainer.getBoundingClientRect();
+  const pageRect = pageContainer.getBoundingClientRect();
+
+  const offsetX = Math.ceil(pageRect.x - canvasRect.x);
+  const offsetY = Math.floor(pageRect.y - canvasRect.y);
+
+  const verticalGuidelines = canSnap
+    ? [offsetX, offsetX + canvasWidth / 2, offsetX + canvasWidth]
```

Comment:
This seems to be based on the safezone but not the fullbleed -- meaning that the snaplines would shown only on the safezone, shouldn't the fullbleed snaplines still remain as well for being able to snap to the edges? Especially when the `showSafeZone` is set to `false`? Couldn't we use `fullbleedContainer` here?

Reply:
Hmm, generally curious why the vertical / horizontal guidelines needed changing.

Reply:
Because they snapped to the top left corner of the workspace! They've been broken for a long time - it just wasn't noticed because we also had the bg element as a snap target.

What else would these weird phantom snaplines be (deus ex machina)?

<img width="595" alt="Screen Shot 2020-09-28 at 11 48 23" src="https://user-images.githubusercontent.com/637548/94455121-b8097000-0180-11eb-940b-b3df7c718e6d.png">

And yes, it makes sense to snap to the fullbleed element if safe mode is off.

Reply:
> Because they snapped to the top left corner of the workspace! 

Hah, okay! I thought the PR changed the snapping from full-bleed to safe zone only but if it was snapping to the workspace initially then that's another story.